apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/DslJsonSerializer.java

```diff
@@ -677,9 +677,14 @@ private void serializeStackTraceElement(StackTraceElement stacktrace) {
         jw.writeByte(OBJECT_START);
         writeField("filename", stacktrace.getFileName());
         writeField("function", stacktrace.getMethodName());
-        writeField("library_frame", isLibraryFrame(stacktrace.getClassName()));
+        String className = stacktrace.getClassName();
+        writeField("library_frame", isLibraryFrame(className));
         writeField("lineno", stacktrace.getLineNumber());
-        serializeStackFrameModule(stacktrace.getClassName());
+        int lastDotIndex = className.lastIndexOf('.');
+        if (lastDotIndex > 0) {
+            writeField("abs_path", className.substring(0, lastDotIndex).replace('.', '/') + "/" + stacktrace.getFileName());
```

Comment:
This allocates two String objects (`substring()` and `replace()`) for each stack frame of each span. Can we use a char-copy to a StringBuilder that includes the char replacement as well?
If so, can it also be used for handling inner classes, lambdas and the like or that using `getFileName()` will always be safer?